rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/repository/EvaluationRepositoryImpl.java

```diff
@@ -133,4 +134,21 @@ public int deleteByOwnerId(String ownerId) {
         evaluations.forEach(session::delete);
         return evaluations.size();
     }
+
+
+    public void deleteByList(List<Evaluation> evaluations) {
+        Session session = sessionFactory.getCurrentSession();
+        evaluations.forEach(eval -> session.delete((Evaluation) session.merge(eval)));
```

Comment:
instead of merging the entity and then deleting you could use deleteById which generally incurs less overhead.

Reply:
I replaced the method with deleteById